### PR TITLE
Allow SO_REUSEPORT on http inbound

### PR DESCRIPTION
**Pull request by breerly**
@yarpc/python

### CODE REVIEW

File: yarpc/transport/http/inbound.py [hunk 1]
```diff
@@ -33,9 +34,14 @@
...
 
-    def __init__(self, port, handlers=None):
+    def __init__(self, port, reuse_port=False, handlers=None):
         self._port = port
+        if reuse_port is True:
+            self._reuse_port = True
+        else:
+            self._reuse_port = False
         self._handlers = handlers or []
+        self._http_server = None
```
Comment (breerly) [inline yarpc/transport/http/inbound.py]: whoops

**Comment by junchaowu** [comment]
make sure you don't reuse port between http and tchannel

**Comment by breerly** [comment]
@junchaowu I don't want the HTTP inbound to know anything about the other inbounds

File: yarpc/transport/http/inbound.py [hunk 2]
```diff
@@ -33,8 +34,12 @@
 
 class HTTPInbound(object):
 
-    def __init__(self, port, handlers=None):
+    def __init__(self, port, reuse_port=False, handlers=None):
         self._port = port
+        if reuse_port is True:
+            self._reuse_port = True
+        else:
+            self._reuse_port = False
```
Comment (abhinav) [inline yarpc/transport/http/inbound.py]: why not

    self._reuse_port = reuse_port

**Comment by abhinav** [comment]
:+1:

**Pull request merged**